sphinx_gallery/gen_rst.py

```diff
@@ -709,7 +711,19 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
 
     time_m, time_s = divmod(time_elapsed, 60)
     example_nb = jupyter_notebook(script_blocks)
-    save_notebook(example_nb, replace_py_ipynb(example_file))
+    path_ipynb = replace_py_ipynb(example_file)
+    save_notebook(example_nb, path_ipynb)
+
+    # Copy the ipynb file to the static folder, preserving folder structure
+    if binder_conf.get('notebooks_folder'):
```

Comment:
I don't like doing the copy part here nor passing the app object this deep, this breaks my desire of making sphinx-gallery more modular. This copy task should happen at the end of the build in the same way we implement `embed_code_links`.
That is have in setup something like this
```python
app.connect('build-finished',copy_jupyter_nb)
```
And then in binder.py the funtion that does the copy.

Reply:
maybe for the the copy just use an `os.walk` and copy everything that is a notebook.

Reply:
yeah I think os.walk is the path forward here...lemme give a shot at modularizing this a little bit more.